node/network/dispute-distribution/src/lib.rs

```diff
@@ -247,9 +272,10 @@ impl MuxedMessage {
 		// ends.
 		let from_overseer = ctx.recv().fuse();
 		futures::pin_mut!(from_overseer, from_sender);
-		futures::select!(
-			msg = from_overseer => MuxedMessage::Subsystem(msg.map_err(FatalError::SubsystemReceive)),
+		// We select biased to make sure we finish up loose ends, before starting new work.
+		futures::select_biased!(
```

Comment:
Not sure about that. Shouldn't signals like `Finalized` be prioritized so we don't waste work on disputes about finalized blocks?

Reply:
We don't [stop any work on finalization](https://github.com/paritytech/polkadot/blob/f9fc6c885688c870759d27b2f56318a1f066e9bf/node/network/dispute-distribution/src/lib.rs#L237), but only when disputes become inactive. Disputes are allowed to carry on once the block is finalized, to account for "late" votes.